src/Microsoft.DotNet.Build.Tasks/PackageFiles/buildagainstpackages.targets

```diff
@@ -19,32 +21,68 @@
       DetermineProjectJsonPath;
       ResolveNuGetPackages;
     </ResolveAssemblyReferencesDependsOn>
+    <ResolveNuGetPackagesDependsOn>
+      $(ResolveNuGetPackagesDependsOn);
+      DetermineProjectJsonPath;
+    </ResolveNuGetPackagesDependsOn>
   </PropertyGroup>
 
   <Target Name="GenerateTestProjectJson" DependsOnTargets="ExtractProjectReferences;GatherAssemblyVersionsFromPackagesDrops;GatherProjectReferenceForProjectJson;DetermineProjectJsonPath;AddDependenciesToProjectJson" />
-
-  <!-- Override targets for building against packages -->
-  <Target Name="BuildAndTest" DependsOnTargets="GenerateTestProjectJson;Build;Test" />
-  <Target Name="RebuildAndTest" DependsOnTargets="GenerateTestProjectJson;Rebuild;Test" />
-
-  <Target Name="DetermineProjectJsonPath">
+  
+  <Target Name="DetermineProjectJsonPath"
+          DependsOnTargets="EvaluateBuildsProjects">
     <PropertyGroup>
       <OriginalProjectJson>$(ProjectJson)</OriginalProjectJson>
+      <GeneratedTargetGroup Condition="'$(GeneratedTargetGroup)' == ''">$(TargetGroup)</GeneratedTargetGroup>
+      <GeneratedOSGroup Condition="'$(GeneratedOSGroup)' == ''">$(OSGroup)</GeneratedOSGroup>
       <!-- We can't put the generated project.json in the output path or NuGet will interpret Microsoft.Win32.Primitives as a "project"
            type instead of a "package" type when generating the project.lock.json file, so we have to place our project.json in a different path location. -->
-      <ProjectJson>$(GeneratedProjectJsonDir)\$(MSBuildProjectName)\json\project.json</ProjectJson>
-      <GeneratedProjectLockJson>$(GeneratedProjectJsonDir)\$(MSBuildProjectName)\json\project.lock.json</GeneratedProjectLockJson>
+      <ProjectsGeneratedProjectJsonDir>$(GeneratedProjectJsonDir)\$(MSBuildProjectName)\</ProjectsGeneratedProjectJsonDir>
+      <ProjectsGeneratedProjectJsonDir Condition="'$(GeneratedOSGroup)' != ''">$(ProjectsGeneratedProjectJsonDir)$(GeneratedOSGroup)\</ProjectsGeneratedProjectJsonDir>
+      <ProjectsGeneratedProjectJsonDir Condition="'$(GeneratedTargetGroup)' != ''">$(ProjectsGeneratedProjectJsonDir)$(GeneratedTargetGroup)\</ProjectsGeneratedProjectJsonDir>
+      <ProjectJson>$(ProjectsGeneratedProjectJsonDir)project.json</ProjectJson>
+      <GeneratedProjectLockJson>$(ProjectsGeneratedProjectJsonDir)project.lock.json</GeneratedProjectLockJson>
       <!-- Not all projects generate a new project.json, so only update the project.lock.json if it has been restored. -->
       <ProjectLockJson Condition="Exists('$(GeneratedProjectLockJson)')">$(GeneratedProjectLockJson)</ProjectLockJson>
     </PropertyGroup>
     <Message Condition="Exists('$(ProjectLockJson)')" Text="Using '$(ProjectLockJson)'" Importance="Low"/>
   </Target>
 
+  <!-- The msbuild project file may be an ".builds" file, if it is, then we need to evaluate the "Project" items in the builds file
+       to generate the project.json's for each of those items. -->
+  <Target Name="EvaluateBuildsProjects"
```

Comment:
When using sync.cmd or build.cmd or build-tests.cmd, in dir.traversal.targets, we end up calling GenerateTestProjectJson for each of the test ".builds" files.  When we try to determine the generated project.json path, we have to then iterate on the proj files described in the ".builds" files, so we call "EvaluateBuildsProjects".  

When building a project directly, we don't end up calling EvaluteBuildsProjects from buildagainstpackages.targets (because of the Condition in EvaluateBuildsProjects).  

Essentially, EvaluateBuildsProjects is used to iterate on the ".builds" files project files.